index.js

```diff
@@ -20,333 +23,289 @@ const CSOBPaymentModule = class CSOBPaymentModule {
-          reject(Error('Init - Verification failed'));
-        }
-      }).catch(error => {
-        reject(error);
-      });
+  async init(payload) {
+    payload['signature'] = this._sign(this._createPayloadMessage(payload))
+    this.logger('init', payload)
+    const result = await request({
+      url: `${this.config.gateUrl}/payment/init`,
+      method: 'POST',
+      json: true,
+      body: payload
     })
-  };
+    if (this._verify(this._createResultMessage(result), result.signature)) {
+      if (result.resultCode.toString() === '0') {
+        return result
+      } else {
+        throw result
+      }
+    }
+    throw new Error('Init - Verification failed')
+
+  }
 
   // process - 2.krok - redirect url
   getRedirectUrl(id) {
-    const dttm = this._createDttm();
+    const dttm = this._createDttm()
     const signature = this._sign(this._createMessageString({
       merchantId: this.config.merchantId,
       payId: id,
       dttm
     }))
-    const url = `${this.config.gateUrl}/payment/process/${this.config.merchantId}/${id}/${dttm}/${encodeURIComponent(signature)}`;
-    this.logger('redirectUrl', url);
-    return url;
+    const url = `${this.config.gateUrl}/payment/process/${this.config.merchantId}/${id}/${dttm}/${encodeURIComponent(signature)}`
+    this.logger('redirectUrl', url)
+    return url
   }
 
   // status
-  status(id) {
-    return new RSVP.Promise((resolve, reject) => {
-      const dttm = this._createDttm();
-      const signature = this._sign(this._createMessageString({
-        merchantId: this.config.merchantId,
-        payId: id,
-        dttm
-      }))
-
-      const url = `${this.config.gateUrl}/payment/status/${this.config.merchantId}/${id}/${dttm}/${encodeURIComponent(signature)}`;
-      this.logger('status', url);
-      rp({
-        url,
-        method: "GET",
-        json: true
-      }).then(result => {
-        const message = this._createResultMessage(result);
-
-        if (this._verify(message, result.signature)) {
-          if (result.resultCode.toString() === '0') {
-            resolve(result)
-          } else {
-            reject(result);
-          }
-        } else {
-          reject(Error('Status - Verification failed'));
-        }
-      }).catch(error => {
-        reject(error);
-      });
+  async status(id) {
+    const dttm = this._createDttm()
+    const signature = this._sign(this._createMessageString({
+      merchantId: this.config.merchantId,
+      payId: id,
+      dttm
+    }))
+
+    const url = `${this.config.gateUrl}/payment/status/${this.config.merchantId}/${id}/${dttm}/${encodeURIComponent(signature)}`
+    this.logger('status', url)
+    const result = request({
+      url,
+      method: 'GET',
+      json: true
     })
+    const message = this._createResultMessage(result)
+
+    if (this._verify(message, result.signature)) {
+      if (result.resultCode.toString() === '0') {
+        return result
+      } else {
+        throw result
```

Comment:
?

Reply:
dont approve if you see something weird :)